### PR TITLE
added access token authenticator

### PR DESCRIPTION
**Pull request by chrisderock**
in my project we use a specialized service to manage oauth2-tokens so i added the AccessTokenAuthenticator for use with plain access tokens. may be useful for others.

### CODE REVIEW

File: src/access_token.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+use crate::error::Error;
+use crate::types::TokenInfo;
+use hyper::client::connect::Connection;
+// use hyper::header;
+use http::Uri;
+// use serde::{Deserialize, Serialize};
+use std::error::Error as StdError;
+use tokio::io::{AsyncRead, AsyncWrite};
+use tower_service::Service;
+// use url::form_urlencoded;
```
Comment (dermesser) [inline src/access_token.rs]: Can you remove the commented imports?

File: src/access_token.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+//! pseudo authenticator for use with plain access tokens
```
Comment (dermesser) [inline src/access_token.rs]: Would you mind expanding a bit on the intended use? (e.g., testing)

**Comment by dermesser** [comment]
Looks good otherwise, simple enough.

File: src/access_token.rs (hunk 1 above)
Comment (chrisderock) [reply src/access_token.rs]: done ;)

File: src/access_token.rs (hunk 2 above)
Comment (chrisderock) [reply src/access_token.rs]: done. maybe some day i can publish the service mentioned but atm its impossible

**Pull request merged**